modules/dnn/CMakeLists.txt

```diff
@@ -295,6 +295,13 @@ if(TARGET ocv.3rdparty.cann AND OPENCV_TEST_DNN_CANN)
   endif()
 endif()
 
+ocv_option(OPENCV_TEST_DNN_TIMVX "Build test with TIM-VX" (TARGET ocv.3rdparty.timvx))
+if(TARGET ocv.3rdparty.timvx AND OPENCV_TEST_DNN_TIMVX)
+  if(TARGET opencv_test_dnn)
+    ocv_target_link_libraries(opencv_test_dnn ocv.3rdparty.timvx)
```

Comment:
Why do you need it? It's internal detail of implementation and user code should not depend on it, including tests. libopencv_dnn should have dependency from TIMVX.

Reply:
Test does not work with missing `HAVE_TIMVX` macro:

https://github.com/opencv/opencv/blob/240b245105bdb69b2b624ce24f08373c2b16bfde/modules/dnn/test/test_int8_layers.cpp#L11-L22

Basically I need `HAVE_TIMVX=1`.

Reply:
We don't need that.

There is registry API:

https://github.com/opencv/opencv/blob/240b245105bdb69b2b624ce24f08373c2b16bfde/modules/dnn/include/opencv2/dnn/dnn.hpp#L125-L126

And usage in tests:

https://github.com/opencv/opencv/blob/240b245105bdb69b2b624ce24f08373c2b16bfde/modules/dnn/test/test_common.impl.hpp#L291-L296

Reply:
`DNN_BACKEND_TIMVX` is not in opencv/modules/dnn/test/test_common.impl.hpp.